services/containerregistry/registries.go

```diff
@@ -439,22 +429,24 @@ func SetTokenData(d *schema.ResourceData, tokenProps cr.TokenProperties) error {
 	return nil
 }
 
-//nolint:golint
-func SetCredentials(credentials cr.Credentials) map[string]interface{} {
+// SetCredentials sets the credentials for the registry token. username can be set on read, password will be set only on create
+func SetCredentials(credentials cr.Credentials) map[string]any {
 
-	credentialsEntry := map[string]interface{}{}
-
-	utils.SetPropWithNilCheck(credentialsEntry, "username", credentials.Username)
+	credentialsEntry := map[string]any{}
+	credentialsEntry["username"] = credentials.Username
+	if credentials.Password != "" {
```

Comment:
@cristiGuranIonos this check is necessary because the `password` is returned only on `POST` requests, I assume that for `GET` requests, the password will be `""`, right?

Reply:
yes, but the function is used in the data source which does not have the `password` field and would crash on write.